packages/common-server/src/etc.ts

```diff
@@ -61,7 +61,13 @@ export class WebViewCommonUtils {
 
     <script type="text/javascript">
       var theme = 'unknown';
-
+      
+      document.addEventListener('copy', (e) => {
```

Comment:
can you fix the formatting here to be in line with the rest of the doc?

Reply:
Will need to circle back to this one at another time thus far not going far.

# Some further attempts
Copying preview of:
```
## XVault Link

- vault1 bar: [[dendron://vault/bar]]
- vault2 bar: [[dendron://vault2/bar]]

## URL
[[https://wiki.dendron.so/#getting-started]]
```

## In JS
In JS with the following code:
```
      document.addEventListener('copy', (e) => {

          console.log("--------------------------------------------------------------------------------");
          console.log("---window.getSelection().toString()---");
          console.log(window.getSelection().toString());

          console.log("---JSON.stringify(window.getSelection())---");
          console.log(JSON.stringify(window.getSelection()));

          console.log("---e.clipboardData.getData('text/html')---");
          console.log(e.clipboardData.getData('text/html'));

          console.log("---window.clipboard.readText()---");
          if (window.clipboard === undefined){
               console.log("window.clipboard is UNDEFINED");
          }else{
              console.log(window.clipboard.readText());
          }
          console.log("--------------------------------------------------------------------------------");

          window.vscode.postMessage({
              type: 'onCopy',
          });
      });
```


```
--------------------------------------------------------------------------------
---window.getSelection().toString()---
XVault Link
vault1 bar: bar
vault2 bar: Bar
URL
https://wiki.dendron.so/

---JSON.stringify(window.getSelection())---
{}

---e.clipboardData.getData('text/html')---

---window.clipboard.readText()---
window.clipboard is UNDEFINED
--------------------------------------------------------------------------------
```

## Other APIs

Looking around the APIs this API sounds like it should return more than plain text:
https://developer.mozilla.org/en-US/docs/Web/API/Clipboard/read

However, VSCode interface for clipboard appears only to have read text for read:
```
    export interface Clipboard {

        /**
         * Read the current clipboard contents as text.
         * @returns A thenable that resolves to a string.
         */
        readText(): Thenable<string>;

        /**
         * Writes text into the clipboard.
         * @returns A thenable that resolves when writing happened.
         */
        writeText(value: string): Thenable<void>;
}
```

## Trying navigator API

```

      document.addEventListener('copy', async (e) => {

          console.log("COPY EVENT--------------------------------------------------------------------------------");
          const clipboardItems = await window.navigator.clipboard.read();

          console.log(clipboardItems.length);
          console.log(clipboardItems);
          console.log(JSON.stringify(clipboardItems));

          console.log("--------------------------------------------------------------------------------");

          window.vscode.postMessage({
              type: 'onCopy',
          });
      });

        document.addEventListener('paste', async (e) => {

          console.log("PASTE EVENT--------------------------------------------------------------------------------");
          const clipboardItems = await window.navigator.clipboard.read();
          console.log(clipboardItems.length);
          console.log(clipboardItems);
          console.log(JSON.stringify(clipboardItems));

          console.log("--------------------------------------------------------------------------------");


      });
```

Output
```
COPY EVENT--------------------------------------------------------------------------------
1
[ClipboardItem]
[{}]
--------------------------------------------------------------------------------
PASTE EVENT--------------------------------------------------------------------------------
1
[ClipboardItem]
[{}]
--------------------------------------------------------------------------------
```
When browsing the content of Clipboard item not able to see the data that was copied.

Reply:
1. can you set a `debugger` in the webview to see if there are other properties of the event that have the output? see [[Qa|dendron://dendron.docs/pkg.dendron-plugin-views.qa]]

2. there might be more information in the following places
- [Clipboard VSCOde Issues · microsoft/vscode](https://github.com/microsoft/vscode/labels/editor-clipboard): find clipboard issues might provide more clues
    - [Add setting to keep tab characters in HTML clipboard · Issue #134588 · microsoft/vscode](https://github.com/microsoft/vscode/issues/134588)
    - [Expose Clipboard#read/write with data transfer · Issue #77790 · microsoft/vscode](https://github.com/microsoft/vscode/issues/77790)
- [vscode-paste-special/extension.ts at master · d3v-one/vscode-paste-special](https://github.com/d3v-one/vscode-paste-special/blob/master/src/extension.ts): extension that uses native commands to update clipboard settings

3. if you don't find an adequate solution, can you submit an issue to vscode?

Reply:
In the new commit: tried to approach it by taking the selection on copy event and taking just the HTML.

Website Links appear to work as links and headers appear as headers (when header and something is copied). 

The background is removed.

WikiLinks to note as expected will have nothing to link. They are copied as what appears to be an empty link (retains correct name of the link though).